<a href="https://colab.research.google.com/github/adamarispadilla/final-project/blob/main/Padilla_HW_6_Naive_Bayes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Brain Tumor CSV file

**# Class column defines either the image has tumor or not (0 = Tumor, 1 = Non-Tumor)**


In [ ]:
##importing required libraries 
import pandas as pd
import numpy as np

df=pd.read_csv("Brain Tumor.csv")
df

# Class column defines either the image has tumor or not (0 = Tumor, 1 = Non-Tumor)


Image  Class       Mean     Variance  Standard Deviation   Entropy  \
0        Image1      0   6.535339   619.587845           24.891522  0.109059   
1        Image2      0   8.749969   805.957634           28.389393  0.266538   
2        Image3      1   7.341095  1143.808219           33.820234  0.001467   
3        Image4      1   5.958145   959.711985           30.979219  0.001477   
4        Image5      0   7.315231   729.540579           27.010009  0.146761   
...         ...    ...        ...          ...                 ...       ...   
3757  Image3758      0  21.234512  1208.850174           34.768523  0.063774   
3758  Image3759      0  20.435349  1227.151440           35.030721  0.066763   
3759  Image3760      0  18.011520  1151.582765           33.934978  0.068396   
3760  Image3761      0  13.330429   945.732779           30.752769  0.087872   
3761  Image3762      0   6.110138   480.884025           21.929068  0.118171   

      Skewness   Kurtosis    Contrast    Energy       ASM  Homogeneity  \
0     4.276477  18.900575   98.613971  0.293314  0.086033     0.530941   
1     3.718116  14.464618   63.858816  0.475051  0.225674     0.651352   
2     5.061750  26.479563   81.867206  0.031917  0.001019     0.268275   
3     5.677977  33.428845  151.229741  0.032024  0.001026     0.243851   
4     4.283221  19.079108  174.988756  0.343849  0.118232     0.501140   
...        ...        ...         ...       ...       ...          ...   
3757  2.082079   4.647310  158.437600  0.220666  0.048693     0.487131   
3758  2.144625   4.882034  161.158675  0.225931  0.051045     0.502712   
3759  2.308349   5.579498  167.130118  0.228930  0.052409     0.492269   
3760  2.732822   7.757570  223.812932  0.261527  0.068397     0.480064   
3761  4.110669  17.538826  239.251388  0.306224  0.093773     0.494333   

      Dissimilarity  Correlation     Coarseness  
0          4.473346     0.981939  7.460000e-155  
1          3.220072     0.988834  7.460000e-155  
2          5.981800     0.978014  7.460000e-155  
3          7.700919     0.964189  7.460000e-155  
4          6.834689     0.972789  7.460000e-155  
...             ...          ...            ...  
3757       5.211739     0.950972  7.460000e-155  
3758       5.083126     0.952749  7.460000e-155  
3759       5.103700     0.952181  7.460000e-155  
3760       6.439784     0.940898  7.460000e-155  
3761       6.787329     0.938731  7.460000e-155  

[3762 rows x 15 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3762 entries, 0 to 3761
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Image               3762 non-null   object 
 1   Class               3762 non-null   int64  
 2   Mean                3762 non-null   float64
 3   Variance            3762 non-null   float64
 4   Standard Deviation  3762 non-null   float64
 5   Entropy             3762 non-null   float64
 6   Skewness            3762 non-null   float64
 7   Kurtosis            3762 non-null   float64
 8   Contrast            3762 non-null   float64
 9   Energy              3762 non-null   float64
 10  ASM                 3762 non-null   float64
 11  Homogeneity         3762 non-null   float64
 12  Dissimilarity       3762 non-null   float64
 13  Correlation         3762 non-null   float64
 14  Coarseness          3762 non-null   float64
dtypes: float64(13), int64(1), object(1)
memory usage: 441.0

In [ ]:
df.shape

(3762, 15)

In [ ]:
##dropping any null values just to double-check
df = df.dropna()
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3762 entries, 0 to 3761
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Image               3762 non-null   object 
 1   Class               3762 non-null   int64  
 2   Mean                3762 non-null   float64
 3   Variance            3762 non-null   float64
 4   Standard Deviation  3762 non-null   float64
 5   Entropy             3762 non-null   float64
 6   Skewness            3762 non-null   float64
 7   Kurtosis            3762 non-null   float64
 8   Contrast            3762 non-null   float64
 9   Energy              3762 non-null   float64
 10  ASM                 3762 non-null   float64
 11  Homogeneity         3762 non-null   float64
 12  Dissimilarity       3762 non-null   float64
 13  Correlation         3762 non-null   float64
 14  Coarseness          3762 non-null   float64
dtypes: float64(13), int64(1), object(1)
memory usage: 470.2

In [ ]:
#rows and columns after dropping the columns with null values 
df.shape

(3762, 15)

In [ ]:
#In order to process we must turn those categorical values into numberic values
#Import LabelEncoder
from sklearn import preprocessing

#creating labelEncoder
le = preprocessing.LabelEncoder()

#Using .fit_transform function to fit label
#encoder and return encoded label
Skewness_label=le.fit_transform(df['Skewness'])

#printing skewness_label
Skewness_label

array([2565, 2142, 2965, ...,  426, 1067, 2452])

In [ ]:
#repeat it for Variance label
Variance_label=le.fit_transform(df['Variance'])

#printing Variance_label
Variance_label

array([1837, 2433, 3060, ..., 3067, 2746, 1351])

In [ ]:
#repeat it for class label
class_label=le.fit_transform(df['Class'])

#printing class_label
class_label

array([0, 0, 1, ..., 0, 0, 0])

In [ ]:
#we'll use zip method in this function called merge to combine the two labels: Skewness and Variance
def merge (list1, list2):
  merged_list=tuple(zip(list1, list2))
  return merged_list

features=merge(Skewness_label, Variance_label)
features


((2565, 1837),
 (2142, 2433),
 (2965, 3060),
 (3149, 2759),
 (2569, 2221),
 (2152, 1801),
 (1288, 2707),
 (722, 2940),
 (150, 3082),
 (2612, 2984),
 (2265, 3010),
 (2057, 3065),
 (1783, 2456),
 (1571, 1097),
 (2616, 1162),
 (1601, 2803),
 (1186, 2953),
 (2221, 1272),
 (2186, 838),
 (1462, 1336),
 (1573, 699),
 (3212, 61),
 (2479, 657),
 (1863, 1035),
 (1052, 1838),
 (2615, 2319),
 (3471, 69),
 (2541, 1137),
 (1774, 1608),
 (313, 2640),
 (2207, 433),
 (1669, 2279),
 (1628, 2437),
 (3282, 305),
 (2687, 76),
 (2348, 97),
 (1911, 418),
 (2256, 177),
 (286, 378),
 (730, 323),
 (1527, 184),
 (3216, 30),
 (1792, 1558),
 (1053, 2376),
 (273, 2878),
 (44, 2958),
 (2556, 348),
 (2620, 1596),
 (2357, 1712),
 (1519, 2314),
 (605, 2988),
 (1158, 3392),
 (1417, 3439),
 (1694, 3340),
 (2938, 2401),
 (2482, 432),
 (3570, 330),
 (3145, 996),
 (3009, 1547),
 (1647, 1366),
 (804, 796),
 (2618, 197),
 (2725, 740),
 (2136, 1118),
 (383, 2734),
 (1126, 1601),
 (2325, 514),
 (2209, 2558),
 (3637, 315),
 (36,

In [ ]:
#Import Gaussian Naive Bayes model
from sklearn.naive_bayes import GaussianNB

#Create a Gaussian Classifier
model = GaussianNB()

# Train the model using the training sets
model.fit(features, class_label)

#Predict 
predicted= model.predict([[2565, 1837]]) ##2565 = 4.276477026 skewness , 1837 = 619.5878446 varience 
print ("Predicted Value:", predicted)

##The model predicts that the person has a tumor. 

Predicted Value: [1]


After encoding the Skewness and Varience levels of the image, we get a set of numbers. We use those numbers in the Guassian Classifier. We plugged in 2565 (4.3 skewness), 1837 (619.6 varience) which resulted in 1. This indicates whether the person does not have a tumor based on the image scan. The following values indicate whether a person has a tumor or not.

1 = No Tumor 

0 = Tumor



**ACCURACY**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3762 entries, 0 to 3761
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Image               3762 non-null   object 
 1   Class               3762 non-null   int64  
 2   Mean                3762 non-null   float64
 3   Variance            3762 non-null   float64
 4   Standard Deviation  3762 non-null   float64
 5   Entropy             3762 non-null   float64
 6   Skewness            3762 non-null   float64
 7   Kurtosis            3762 non-null   float64
 8   Contrast            3762 non-null   float64
 9   Energy              3762 non-null   float64
 10  ASM                 3762 non-null   float64
 11  Homogeneity         3762 non-null   float64
 12  Dissimilarity       3762 non-null   float64
 13  Correlation         3762 non-null   float64
 14  Coarseness          3762 non-null   float64
dtypes: float64(13), int64(1), object(1)
memory usage: 470.2

In [ ]:
list(df.columns)

['Image',
 'Class',
 'Mean',
 'Variance',
 'Standard Deviation',
 'Entropy',
 'Skewness',
 'Kurtosis',
 'Contrast',
 'Energy',
 'ASM',
 'Homogeneity',
 'Dissimilarity',
 'Correlation',
 'Coarseness']

In [ ]:
df.colums = [
 'Class',
 'Mean',
 'Variance',
 'Standard Deviation',
 'Entropy',
 'Skewness',
 'Kurtosis',
 'Contrast',
 'Energy',
 'ASM',
 'Homogeneity',
 'Dissimilarity',
 'Correlation',
 'Coarseness']

df.head()

Image  Class      Mean     Variance  Standard Deviation   Entropy  \
0  Image1      0  6.535339   619.587845           24.891522  0.109059   
1  Image2      0  8.749969   805.957634           28.389393  0.266538   
2  Image3      1  7.341095  1143.808219           33.820234  0.001467   
3  Image4      1  5.958145   959.711985           30.979219  0.001477   
4  Image5      0  7.315231   729.540579           27.010009  0.146761   

   Skewness   Kurtosis    Contrast    Energy       ASM  Homogeneity  \
0  4.276477  18.900575   98.613971  0.293314  0.086033     0.530941   
1  3.718116  14.464618   63.858816  0.475051  0.225674     0.651352   
2  5.061750  26.479563   81.867206  0.031917  0.001019     0.268275   
3  5.677977  33.428845  151.229741  0.032024  0.001026     0.243851   
4  4.283221  19.079108  174.988756  0.343849  0.118232     0.501140   

   Dissimilarity  Correlation     Coarseness  
0       4.473346     0.981939  7.460000e-155  
1       3.220072     0.988834  7.460000e-155  
2       5.981800     0.978014  7.460000e-155  
3       7.700919     0.964189  7.460000e-155  
4       6.834689     0.972789  7.460000e-155

In [ ]:
x_data = df[[
 'Mean',
 'Variance',
 'Standard Deviation',
 'Entropy',
 'Skewness',
 'Kurtosis',
 'Contrast',
 'Energy',
 'ASM',
 'Homogeneity',
 'Dissimilarity',
 'Correlation',
 'Coarseness']]
y_data = df['Class']

In [ ]:
# Import train_test_split function
from sklearn.model_selection import train_test_split

# Split dataset into training set and test set, 70% training and 30% test
X_train, X_test, y_train, y_test = train_test_split(x_data, y_data, test_size=0.3,random_state=109) 

In [ ]:
#This is model generation

#Import Gaussian Naive Bayes model
from sklearn.naive_bayes import GaussianNB

#Create a Gaussian Classifier
model = GaussianNB()

#Train the model using the training sets
model.fit(X_train, y_train)

#Predict the response for test dataset
y_pred = model.predict(X_test)

#This is to evaluate the model
#Import scikit-learn metrics module for accuracy calculation
from sklearn import metrics

# Model Accuracy, how often is the classifier correct?
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))

Accuracy: 0.9636846767050488


**The classifier is correct 96% of the time.**